ql/src/test/org/apache/hadoop/hive/ql/udf/generic/TestGenericUDFDateFormat.java

```diff
@@ -166,7 +166,7 @@ public void testNullFmt() throws HiveException {
   public void testWrongFmt() throws HiveException {
     GenericUDFDateFormat udf = new GenericUDFDateFormat();
     ObjectInspector valueOI0 = PrimitiveObjectInspectorFactory.writableStringObjectInspector;
-    Text fmtText = new Text("B");
+    Text fmtText = new Text("C");
```

Comment:
why change this?

Reply:
'B' is defined as a pattern in jdk17 to represent period-of-day, it has been changed to pattern 'C' having no definition in jdk17 which is the requirement of test case here.

Reply:
we should document all visible changes in jdk-17 migration guide/doc